### PR TITLE
fix: address shellcheck warnings

### PR DESCRIPTION
**Pull request by lucaspin**
We run [shellcheck](https://github.com/koalaman/shellcheck) during our CI, but we don't do anything about it. We should. This pull request addressed all the warnings for `checkout`, `sem-service` and `sem-version`. The main complaints were:
- https://www.shellcheck.net/wiki/SC2068
- https://www.shellcheck.net/wiki/SC2086
- https://www.shellcheck.net/wiki/SC2236
- https://www.shellcheck.net/wiki/SC2181
- https://www.shellcheck.net/wiki/SC2164
- https://www.shellcheck.net/wiki/SC2124
- https://www.shellcheck.net/wiki/SC2128
- https://www.shellcheck.net/wiki/SC2145
- https://www.shellcheck.net/wiki/SC2034

### CODE REVIEW

File: libcheckout [hunk 1]
```diff
@@ -49,6 +48,7 @@ function checkout::use_cache() {
     checkout::clone
   fi
 
+  # shellcheck disable=SC2181
```
Comment (radwo) [inline libcheckout:51]: Can you put an explanation why we are turning off checks?

**Review by radwo — COMMENTED** [review]
Do we need to report checks for each block, or can we switch to the [pipeline level](https://docs.semaphoreci.com/essentials/configuring-status-checks/)?
We will be able to set up branch protection rules.

**Comment by lucaspin** [comment]
@radwo 

> Do we need to report checks for each block, or can we switch to the pipeline level?

We can switch to reporting checks on the pipeline level.

File: libcheckout (hunk 1 above)
Comment (lucaspin) [reply libcheckout:53]: Done.

**Review by radwo — APPROVED** [review]

**Pull request merged**